packages/plugin-core/src/commands/AddExistingVaultCommand.ts

```diff
@@ -0,0 +1,504 @@
+            selected.label === "custom"
+              ? GitUtils.getRepoNameFromURL(sourceRemotePath)
+              : selected.label;
+          const placeHolder = path2Vault;
+          sourceName = await VSCodeUtils.showInputBox({
+            prompt: "Name of new vault (optional, press enter to skip)",
+            value: placeHolder,
+          });
+          qp.hide();
+          return resolve({
+            type: sourceType!,
+            name: sourceName,
+            path: vaultDestination,
+            pathRemote: sourceRemotePath,
+          });
+        });
+        qp.show();
+      });
+      return out;
+    }
+    const placeHolder = path.basename(vaultDestination);
+    sourceName = await VSCodeUtils.showInputBox({
+      prompt: "Name of new vault (optional, press enter to skip)",
+      placeHolder,
+    });
+    return {
+      type: sourceType,
+      name: sourceName,
+      path: vaultDestination,
+    };
+  }
+
+  async gatherDestinationFolder() {
+    const defaultUri = Uri.file(this._ext.getDWorkspace().wsRoot);
+    // Prompt user where to create new vault
+    const options: OpenDialogOptions = {
+      canSelectMany: false,
+      openLabel: "Select vault to add",
+      canSelectFiles: false,
+      canSelectFolders: true,
+      defaultUri,
+    };
+    const folder = await VSCodeUtils.openFilePicker(options);
+    if (_.isUndefined(folder)) {
+      return;
+    }
+    return folder;
+  }
+
+  async gatherVaultSelfContained(
+    sourceType: VaultRemoteSource,
+    vaultDestination: string
+  ): Promise<CommandOpts | undefined> {
+    if (sourceType === VaultType.LOCAL) {
+      const sourceName = await VSCodeUtils.showInputBox({
+        prompt: "Name of new vault (optional, press enter to skip)",
+        placeHolder: path.basename(vaultDestination),
+      });
+
+      return {
+        type: sourceType,
+        name: sourceName,
+        path: vaultDestination,
+        isSelfContained: true,
+      };
+    } else {
+      // Remote vault
+      const remote = await VSCodeUtils.showInputBox({
+        title: "Remote URL",
+        prompt: "Enter the URL for the git remote",
+        placeHolder: "git@github.com:dendronhq/dendron.git",
+        ignoreFocusOut: true,
+      });
+      // Cancelled
+      if (PickerUtilsV2.isInputEmpty(remote)) return;
+
+      // Calculate the vault name from the remote.
+      const vaultName: string | undefined = GitUtils.getRepoNameFromURL(remote);
+
+      const sourceName = await VSCodeUtils.showInputBox({
+        prompt: "Name of new vault (optional, press enter to skip)",
+        value: vaultName,
+      });
+
+      return {
+        type: sourceType,
+        name: sourceName,
+        path: vaultDestination,
+        pathRemote: remote,
+        isSelfContained: true,
+      };
+    }
+  }
+
+  async gatherInputs(): Promise<CommandOpts | undefined> {
+    const sourceTypeSelected = await VSCodeUtils.showQuickPick([
+      { label: VaultType.LOCAL, picked: true },
```

Comment:
can we add a `description` string for both items?

- local: A local vault is a Dendron vault that is present in your computer
- remote: A remote vault is a Dendron vault that is available at a git endpoint

also add a `detail` string to show an example

- local: `eg. /home/dendron/hello-vault`
- remote: `eg. git@github.com:dendronhq/dendron-site.git`